src/modules/api/user/link/link.controller.ts

```diff
@@ -1,7 +1,22 @@
-import { Controller } from '@nestjs/common';
+import { Controller, Get, Param } from '@nestjs/common';
 import { LinkService } from './link.service';
+import { IResult } from 'src/common/interfaces/response';
 
 @Controller('users/:userId/links')
 export class LinkController {
-  constructor(private linkService: LinkService) {}
+  constructor(private readonly linkService: LinkService) {}
+  @Get()
+  async getAllLinksForUserId(@Param('userId') userId: string) {
+    const links = await this.linkService.getAllLinksForUser(userId);
+    const response = {} as IResult;
+    if (links?.length) {
+      response.status = 'OK';
+      response.result = links;
+    } else {
+      response.status = 'NOT OK';
+      response.error = 'No Links Available';
+      response.result = links;
```

Comment:
Can we do it in a different way ? No links available is not an error, but an actual response with an empty list

Reply:
Yeah but however i am returning the empty list

Reply:
What I mean to say is, If I'm a new User, I wont be having any links, so when I fetch my links, I should get an OK response, with an empty list. However, if there's an issue with database, or I'm fetching unauthorised data or any such problems, I should get a NOT OK response with just an error message.

Reply:
Just checking `if(links)` would solve the problem I guess.

Reply:
hey if(links) wontb work as it returns an empty array which is not null and valid as well So i went with length

Reply:
But this will give a `NOT OK` response if I have 0 links right?